### PR TITLE
refactor(protocol-designer): annotate export types in several modules

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR follows up on #5785, #5800, and #5803 by annotating the types of module exports in several PD modules:

- `analytics`
- `dismiss`
- `feature-flags`
- `file-data`
- `labware-defs`
- `labware-ingred`

## changelog

- [ ] Annotate types of exports in the above modules

## review requests

- [ ] Ensure all changes are solely to Flow types so that this PR does not affect runtime logic

## risk assessment

Very low if changes are limited to Flow definitions

### CODE REVIEW

File: protocol-designer/src/labware-defs/actions.js [hunk 1]
```diff
@@ -91,9 +91,11 @@ const getIsOverwriteMismatched = (
   return !(matchedWellOrdering && matchedMultiUse)
 }
 
-const _createCustomLabwareDef = (onlyTiprack: boolean) => (
+const _createCustomLabwareDef: (
+  onlyTiprack: boolean
+) => (
   event: SyntheticInputEvent<HTMLInputElement>
-): ThunkAction<*> => (dispatch: ThunkDispatch<*>, getState: GetState) => {
+) => ThunkAction<any> = onlyTiprack => event => (dispatch, getState) => {
```
Comment (mcous) [inline protocol-designer/src/labware-defs/actions.js:98]: These could also be `ThunkAction<Action>` with a $FlowFixMe if people prefer

File: protocol-designer/src/types.js [hunk 2]
```diff
@@ -36,21 +36,28 @@ export type ThunkDispatch<A> = (action: A | ThunkAction<A>) => A
 export type ThunkAction<A> = (
   dispatch: ThunkDispatch<A>,
   getState: GetState
-) => A
+) => A | void
 
 // TODO(mc, 2018-04-18): make actual Action union type for PD
```
Comment (mcous) [inline protocol-designer/src/types.js:41]: This TODO is still important! I tried to change `ThunkAction<A>` to just `ThunkAction` with PD's `Action` type, but given that `Action` is currently this inexact object, it doesn't work.

If PD were to have an exact Action union, these generics could go away

File: protocol-designer/src/labware-ingred/reducers/index.js [hunk 3]
```diff
@@ -327,7 +340,7 @@ export const ingredLocations = handleActions<LocationsState, *>(
 )
 
 export type RootState = {|
-  modeLabwareSelection: ?DeckSlot,
+  modeLabwareSelection: DeckSlot | false,
```
Comment (mcous) [inline protocol-designer/src/labware-ingred/reducers/index.js:343]: So `DeckSlot | false` is weird but this is what's actually happening in the reducer

Comment (IanLondon) [reply protocol-designer/src/labware-ingred/reducers/index.js:343]: eww

Comment (mcous) [reply protocol-designer/src/labware-ingred/reducers/index.js:343]: Hey it still works just fine though!

I will say that one thing I am appreciating about this whole annotating exports thing is that it's making these sorts of things harder to slip through the cracks, which is always nice

**Review by IanLondon — APPROVED** [review]

**Review by shlokamin — APPROVED** [review]
🚢 Thanks @mcous!

**Review by Kadee80 — APPROVED** [review]
🍅

**Pull request merged**